opensearch_py_ml/ml_commons/rag_pipeline/rag/rag_setup.py

```diff
@@ -0,0 +1,486 @@
+            try:
+                response = self.aoss_client.batch_get_collection(ids=[collection_id])
+                status = response['collectionDetails'][0]['status']
+                if status == 'ACTIVE':
+                    print(f"Collection '{self.collection_name}' is now active.")
+                    return True
+                elif status in ['FAILED', 'DELETED']:
+                    print(f"Collection creation failed or was deleted. Status: {status}")
+                    return False
+                else:
+                    print(f"Collection status: {status}. Waiting...")
+                    time.sleep(30)
+            except Exception as ex:
+                print(f"Error checking collection status: {ex}")
+                time.sleep(30)
+        print(f"Timed out waiting for collection to become active after {max_wait_minutes} minutes.")
+        return False
+
+    def get_collection_endpoint(self):
+        if not self.is_serverless:
+            return self.opensearch_endpoint
+        
+        try:
+            collection_id = self.get_collection_id(self.collection_name)
+            if not collection_id:
+                print(f"Collection '{self.collection_name}' not found.")
+                return None
+            
+            batch_get_response = self.aoss_client.batch_get_collection(ids=[collection_id])
+            collection_details = batch_get_response.get('collectionDetails', [])
+            
+            if not collection_details:
+                print(f"No details found for collection ID '{collection_id}'.")
+                return None
+            
+            self.opensearch_endpoint = collection_details[0].get('collectionEndpoint')
+            if self.opensearch_endpoint:
+                print(f"Collection '{self.collection_name}' has endpoint URL: {self.opensearch_endpoint}")
+                return self.opensearch_endpoint
+            else:
+                print(f"No endpoint URL found in collection '{self.collection_name}'.")
+                return None
+        except Exception as ex:
+            print(f"Error retrieving collection endpoint: {ex}")
+            return None
+
+    def initialize_opensearch_client(self):
+        if not self.opensearch_endpoint:
+            print("OpenSearch endpoint not set. Please run setup first.")
+            return False
+        
+        parsed_url = urlparse(self.opensearch_endpoint)
+        host = parsed_url.hostname
+        port = parsed_url.port or 443
+
+        if self.is_serverless:
+            credentials = boto3.Session().get_credentials()
+            auth = AWSV4SignerAuth(credentials, self.aws_region, 'aoss')
+        else:
+            if not self.opensearch_username or not self.opensearch_password:
+                print("OpenSearch username or password not set. Please run setup first.")
+                return False
+            auth = (self.opensearch_username, self.opensearch_password)
```

Comment:
This looks similar to previous code maybe we can abstract it to be reused as a method?

Reply:
No Longer using this block of code!